rust/datafusion/src/optimizer/filter_push_down.rs

```diff
@@ -214,6 +214,36 @@ fn analyze_plan(plan: &LogicalPlan, depth: usize) -> Result<AnalysisResult> {
             result.break_points.insert(depth, columns);
             Ok(result)
         }
+        LogicalPlan::Extension { node } => {
+            let inputs = node.inputs();
+
+            let mut result = match inputs.len() {
+                0 => AnalysisResult {
+                    break_points: BTreeMap::new(),
+                    filters: BTreeMap::new(),
+                    projections: BTreeMap::new(),
+                },
+                1 => analyze_plan(inputs[0], depth + 1)?,
+                _ => {
+                    return Err(DataFusionError::NotImplemented(
```

Comment:
More generally, I am not sure how / if this filter pushdown logic will work if there are multiple child inputs as no existing LogicalPlan variant his multiple inputs at this time. We will have to handle this when / if we want to support joins as well . For now I returned a "Not Yet Implemented" type error

Reply:
I agree :)